File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-3.js

class Solution {
	/**
	 * @param {TreeNode} root
	 * @return {number[][]}
	 */
	levelOrder(root) {
		if (!root) return [];
		const q = [root];
		const ans = [];
		while (q.length) {
			const level = [];
			const copy = [...q];
			for (const node of copy) {
				level.push(node.val);
				q.shift();
				if (node.left)
					q.push(node.left);
				if (node.right)
					q.push(node.right);
			}
			ans.push([...level]);

		}
		return ans;
	}
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        ans = nums[0]

        left, right = 0, len(nums) - 1
        while left <= right:
            mid = left + (right - left) // 2
            ans = min(nums[mid], ans)
            if nums[mid] <= nums[left]:
                right = mid - 1
            else:
                left = mid + 1

        return ans
